### PR TITLE
[EMCAL-630] DPL component for Raw to Cell conversion

### PR DESCRIPTION
**Pull request by hahassan7**

### CODE REVIEW

File: Detectors/EMCAL/workflow/src/RawToCellConverterSpec.cxx [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    bool first = true;
+    uint16_t currentTrigger = 0;
+    uint32_t currentorbit = 0;
+
+    // loop over all the DMA pages
+    while (rawreader.hasNext()) {
+
+      rawreader.next();
+
+      o2::header::RAWDataHeaderV4 header = rawreader.getRawHeader();
```
Comment (mfasDa) [inline Detectors/EMCAL/workflow/src/RawToCellConverterSpec.cxx:77]: Please use `auto` in order to be compatible with future changes of the raw data header.

File: Detectors/EMCAL/workflow/src/RawToCellConverterSpec.cxx [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+void RawToCellConverterSpec::run(framework::ProcessingContext& ctx)
+{
+  LOG(DEBUG) << "[EMCALRawToCellConverter - run] called";
+  auto dataref = ctx.inputs().get("raw");
+  auto const* emcheader = o2::framework::DataRefUtils::getHeader<o2::emcal::EMCALBlockHeader*>(dataref);
+  if (!emcheader->mHasPayload) {
+    LOG(DEBUG) << "[EMCALCellConverter - run] No more raw data" << std::endl;
+    ctx.services().get<o2::framework::ControlService>().readyToQuit(framework::QuitRequest::Me);
+    return;
+  }
```
Comment (mfasDa) [inline Detectors/EMCAL/workflow/src/RawToCellConverterSpec.cxx:56]: This won't work: The raw to cell converter is subscribing directly to readout. There is no EMCAL block header but only the plain raw data. The whole part needs to be removed.

The EMCAL block header is rather intended to notify other components that there is payload data coming. This was based on the first version of the TPC reconstruction workflow. In such case the component would be responsible to send an EMCAL block header. However there were changes in the handling of the workflow termination in #3534 and #3542 which now uses framework level handling of the end-of-stream, which makes the block header superfluous. The DPL components needs to be adapted.

**Comment by hahassan7** [comment]
The failing test is not due to this commit. I think the pull request can be merged.

**Comment by mfasDa** [comment]
For merging the commit should be squash-merged, keeping only the commit message of the first commit.

**Comment by hahassan7** [comment]
Pull request can be merged now.

**Pull request merged**